src/NUnitFramework/framework/Assert.cs

```diff
@@ -313,8 +314,10 @@ public static void Contains(object expected, ICollection actual)
         /// <param name="testDelegate">A TestDelegate to be executed in Multiple Assertion mode.</param>
         public static void Multiple(TestDelegate testDelegate)
         {
-            TestExecutionContext context = TestExecutionContext.CurrentContext;
-            Guard.OperationValid(context != null, "There is no current test execution context.");
+            if (testDelegate is null) throw new ArgumentNullException(nameof(testDelegate));
+
+            TestExecutionContext context = TestExecutionContext.CurrentContext
+                ?? throw new InvalidOperationException("There is no current test execution context.");
```

Comment:
`Guard.OperationValid` is not recognized as contributing to the null check, so this avoids having to suppress a warning in our own code with `!`.

Reply:
Interesting. I wonder if there is a way to signal that?

Reply:
Wait a minute, I wasn't thinking properly when I said that. We can annotate our `Guard.OperationValid` parameter with `[DoesNotReturnIf(false)]` to achieve the desired effect. I'll fix this with my next PR.

Reply:
Wait, now I remember why I didn't annotate it. I have to declare an internal copy of DoesNotReturnIfAttribute. I can do that though.